data.tf

```diff
@@ -5,3 +5,43 @@ data "aws_availability_zones" "available" {
 data "aws_caller_identity" "current" {}
 
 data "aws_region" "current" {}
+
+data "aws_vpc" "preconfigured" {
+  count = local.create_vpc ? 0 : 1
+  id    = var.vpcId
+}
+
+data "aws_subnets" "private_subnets" {
+  count = local.create_vpc ? 0 : 1
+  filter {
+    name   = "vpc-id"
+    values = [var.vpcId]
+  }
+
+  tags = {
+    purpose = "private"

```

Comment:
This is simply to differentiate easily between public/private subnets. It does not exist with clusters deployed before new code, ie needs updating. For new clusters, it will be added automatically.